cs_tools/sync/postgres/syncer.py

```diff
@@ -0,0 +1,119 @@
+from __future__ import annotations
+
+import pydantic
+import sqlalchemy as sa
+import sqlmodel
+from cs_tools.sync.base import DatabaseSyncer
+from typing import TYPE_CHECKING, Any, Literal ,Optional
+from pydantic_core import PydanticCustomError
+import pathlib
+import logging
+from sqlalchemy.orm import sessionmaker
+from cs_tools import __version__
+from cs_tools.sync import utils as sync_utils
+from cs_tools.sync.base import DatabaseSyncer
+
+# if TYPE_CHECKING:
+from cs_tools.sync.types import TableRows
+
+log = logging.getLogger(__name__)
+
+
+class Postgres(DatabaseSyncer):
+    """
+    Interact with Postgres Database
+    """
+    
+    __manifest_path__ = pathlib.Path(__file__).parent / "MANIFEST.json"
+    __syncer_name__ = "postgres"
+   
+    dbname: str 
+    user:  str 
+    password:  str 
+    host:  str 
+    port: int 
+    schema_ : str = pydantic.Field(default="PUBLIC",alias="schema")
+    authentication: Literal["basic", "key-pair", "sso", "oauth"]
+    log_level: Literal["debug", "info", "warning"] = "warning"
+    secret: Optional[str] = None
+    private_key_path: Optional[pydantic.FilePath] = None
+        
+    @pydantic.field_validator("secret", mode="before")
+    def ensure_auth_secret_given(cls, value: Any, info: pydantic.ValidationInfo) -> Any:
+        if info.data.get("authentication") == "sso" or value is not None:
+            return value
+
+        if info.data.get("authentication") == "basic":
+            raise PydanticCustomError("missing", "Field required, you must provide a password", {"secret": value})
+
+        if info.data.get("authentication") == "oauth":
+            raise PydanticCustomError("missing", "Field required, you must provide an oauth token", {"secret": value})
+    
+        
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        logging.getLogger("postgres").setLevel(self.log_level.upper())
```

Comment:
This option and the associated `log_level` can be removed if the pg-sqlalchemy driver is not noisy. **[I did this in Snowflake](https://github.com/thoughtspot/cs_tools/blob/master/cs_tools/sync/snowflake/syncer.py#L67)** because they are verrrrry noisy.